src/main/java/morabito/salvatore/spauthenticator/service/impl/DigitalSignatureServiceImpl.java

```diff
@@ -126,19 +138,24 @@ private String createSignatureString(String requestTargetParameter, HttpHeaders
 
         var signatureString = new StringBuilder(requestTargetParameter);
         /* If there are multiple instances of the same header field, all header field values associated with the header
-           field MUST be concatenated, separated by a ASCII comma and an ASCII space ,, and used in the order in which
+           field MUST be concatenated, separated by a ASCII comma and an ASCII space, and used in the order in which
            they will appear in the transmitted HTTP message.
          */
         requestHeaders.forEach((key, value) -> signatureString.append("\n")
                 .append(key.toLowerCase())
                 .append(Constants.HEADER_KEY_SEPARATOR)
-                .append(value.stream()
-                        .reduce((s1, s2) -> s1 + ", " + s2)));
+                .append(mergeHeaderValues(value)));
         log.info("Signature string: " + signatureString);
 
         return signatureString.toString();
     }
 
+    private static String mergeHeaderValues(List<String> value) {
+        return value.stream()
+                .reduce((s1, s2) -> s1.toString() + ", " + s2.toString())
```

Comment:
what about `String s = list.stream().map(Object::toString).collect(Collectors.joining(","));` as reported [here](https://stackoverflow.com/questions/24882927/using-streams-to-convert-a-list-of-objects-into-a-string-obtained-from-the-tostr)